panther_hardware_interfaces/src/panther_system_ros_interface.cpp

```diff
@@ -117,24 +181,93 @@ void PantherSystemRosInterface::UpdateMsgErrors(const CANErrors & can_errors)
   driver_state.rear.can_net_err = can_errors.rear_can_net_err;
 }
 
+void PantherSystemRosInterface::InitializeAndPublishIOStateMsg(
+  std::shared_ptr<GPIOControllerInterface> gpio_controller, const float panther_version)
+{
+  auto & io_state = realtime_io_state_publisher_->msg_;
+
+  if (panther_version >= 1.2 - std::numeric_limits<float>::epsilon()) {
+    io_state.aux_power = gpio_controller->IsPinActive(panther_gpiod::GPIOPin::AUX_PW_EN);
+    io_state.charger_connected = gpio_controller->IsPinActive(panther_gpiod::GPIOPin::CHRG_SENSE);
+    io_state.charger_enabled = !gpio_controller->IsPinActive(panther_gpiod::GPIOPin::CHRG_DISABLE);
+    io_state.digital_power = !gpio_controller->IsPinActive(panther_gpiod::GPIOPin::VDIG_OFF);
+    io_state.fan = gpio_controller->IsPinActive(panther_gpiod::GPIOPin::FAN_SW);
+    io_state.power_button = gpio_controller->IsPinActive(panther_gpiod::GPIOPin::SHDN_INIT);
+    io_state.motor_on = gpio_controller->IsPinActive(panther_gpiod::GPIOPin::VMOT_ON);
+  } else {
+    io_state.aux_power = true;
+    io_state.charger_connected = false;
+    io_state.charger_enabled = false;
+    io_state.digital_power = true;
+    io_state.fan = false;
+    io_state.power_button = false;
+    io_state.motor_on = gpio_controller->IsPinActive(panther_gpiod::GPIOPin::MOTOR_ON);
+  }
```

Comment:
After discussion with @KmakD and @pawelirh, we have two options:

1. Overriding the `IsPinActive` method in `GPIOController` so that in PTH 1.2 it returns `gpio_driver_->IsPinActive(pin)`, and in 1.06 it has a switch case that returns the appropriate pin state based on the enum.
1. Implementing a Get method in GPIOControllerInterface that would be called in IsPinActive and behave as follows:
- for version 1.20: `return gpio_driver_->IsPinActive(pin);`
- for version 1.06:
    - if the pin is MOTOR_ON, it would call the IsPinActive method of the driver (as in 1.20),
    - if not, it would return a default pin value defined in some map (switch case still needed).

@macstepien, what is your opinion?

Reply:
I have proposed an initial solution in this commit: https://github.com/husarion/panther_ros/pull/222/commits/666983f2b733ad13acdc5e32ed9a67a2eeec40e1

This solution has not yet been tested on the robot.